clang/lib/Sema/SemaLambda.cpp

```diff
@@ -1036,6 +1036,7 @@ void Sema::ActOnLambdaExpressionAfterIntroducer(LambdaIntroducer &Intro,
   // be dependent, because there are template parameters in scope.
   CXXRecordDecl::LambdaDependencyKind LambdaDependencyKind =
       CXXRecordDecl::LDK_Unknown;
+
```

Comment:
Adding a blank line so I can comment here: I found it interesting that we asserted in line 1033 that `LSI->NumExplicitTemplateParams` is always 0... so the lines in the first branch of the if below are useless?
@cor3ntin

Reply:
This looks like dead code from a past refactor, we should remove it
(template parameters are set later, in  `ActOnLambdaExplicitTemplateParameterList`